File: Java/3. Longest Substring Without Repeating Characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Map<Character, Integer> count = new HashMap<>();
        int start = 0, maxLen = 0;
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (count.containsKey(c))
                count.put(c, count.get(c) + 1);
            else
                count.put(c, 1);
            if (count.get(c) > 1) {
                while (start <= i && count.get(c) > 1) {
                    count.put(s.charAt(start), count.get(s.charAt(start)) - 1);
                    start++;
                }
            }            
            maxLen = Math.max(maxLen, i - start + 1);
        }        
        return maxLen;
    }
}

File: Python/3. Longest Substring Without Repeating Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        counter = set()
        start = 0
        length = 0
        for i in range(len(s)):            
            while s[i] in counter and start <= i:                   
                counter.remove(s[start])
                start += 1       
            counter.add(s[i])
            length = max(length, i - start + 1)
        return length
